godot-core/src/engine/script_instance.rs

```diff
@@ -318,32 +259,276 @@ pub fn create_script_instance<T: ScriptInstance>(rs_instance: T) -> *mut c_void
+    ///     fn get_language(&self) -> Gd<ScriptLanguage> { todo!() }
+    ///     fn on_refcount_decremented(&self) -> bool { todo!() }
+    ///     fn on_refcount_incremented(&self) { todo!() }
+    ///     fn property_get_fallback(&self, _: godot::prelude::StringName) -> Option<godot::prelude::Variant> { todo!() }
+    ///     fn property_set_fallback(_: SiMut<'_, Self>, _: godot::prelude::StringName, _: &godot::prelude::Variant) -> bool { todo!() }
+    /// }
+    /// ```
+    ///
+    /// However we cannot call methods that require `&mut Base`, such as
+    /// [`Node::add_child()`](crate::engine::Node::add_child).
+    ///
+    /// ```compile_fail
+    /// use godot::prelude::*;
+    /// use godot::engine::{ScriptInstance, SiMut, ScriptLanguage, Script};
+    /// use godot::builtin::meta::{MethodInfo, PropertyInfo};
+    /// use godot::sys;
+    ///
+    /// struct ExampleScriptInstance;
+    ///
+    /// impl ScriptInstance for ExampleScriptInstance {
+    ///     type Base = Node;
+    ///
+    ///     fn call(
+    ///         this: SiMut<Self>,
+    ///         method: StringName,
+    ///         args: &[&Variant],
+    ///     ) -> Result<Variant, sys::GDExtensionCallErrorType>{
+    ///         let node = Node::new_alloc();
+    ///         // fails because `add_child` requires a mutable reference.
+    ///         this.base().add_child(node);
+    ///
+    ///         Ok(Variant::nil())
+    ///     }
+    ///
+    ///     fn class_name(&self) -> GString { todo!() }
+    ///     fn set_property(_: SiMut<'_, Self>, _: godot::prelude::StringName, _: &godot::prelude::Variant) -> bool { todo!() }
+    ///     fn get_property(&self, _: godot::prelude::StringName) -> Option<godot::prelude::Variant> { todo!() }
+    ///     fn get_property_list(&self) -> Vec<PropertyInfo> { todo!() }
+    ///     fn get_method_list(&self) -> Vec<MethodInfo> { todo!() }
+    ///     fn is_placeholder(&self) -> bool { todo!() }
+    ///     fn has_method(&self, _: godot::prelude::StringName) -> bool { todo!() }
+    ///     fn get_script(&self) -> &Gd<Script> { todo!() }
+    ///     fn get_property_type(&self, _: godot::prelude::StringName) -> VariantType { todo!() }
+    ///     fn to_string(&self) -> GString { todo!() }
+    ///     fn get_property_state(&self) -> Vec<(godot::prelude::StringName, godot::prelude::Variant)> { todo!() }
+    ///     fn get_language(&self) -> Gd<ScriptLanguage> { todo!() }
+    ///     fn on_refcount_decremented(&self) -> bool { todo!() }
+    ///     fn on_refcount_incremented(&self) { todo!() }
+    ///     fn property_get_fallback(&self, _: godot::prelude::StringName) -> Option<godot::prelude::Variant> { todo!() }
+    ///     fn property_set_fallback(_: SiMut<'_, Self>, _: godot::prelude::StringName, _: &godot::prelude::Variant) -> bool { todo!() }
+    /// }
+    /// ```
+    ///
+    /// For this, use [`base_mut()`](SiMut::base_mut()) instead.
```

Comment:
Also here, this is a massive wall of code. Please just include parts that are directly relevant for the method to be documented -- here `base()`.

Since there is only 1 line different in the code block that fails, please just add that as an inline `//` comment, no need to have an entirely separate block.